### ISSUE TITLE
Inline specification of `enable-error-code` silently ignored

### ISSUE THREAD

**Issue by PeterJCLaw** [python/mypy#11789]
**Bug Report**

Inline enabling (and disabling) of error codes is silently ignored, despite this pattern being common in mypy's own documentation.
I suspect that there are a few other options (where they need some post-processing after raw parsing) would be similarly affected.

**To Reproduce**

Given this example, from the docs:
``` python
# mypy: enable-error-code truthy-bool

class Foo:
    pass
foo = Foo()
# Error: "foo" has type "Foo" which does not implement __bool__ or __len__ so it could always be true in boolean context
if foo:
    ...
```
The inline commented error does not in fact appear.

**Expected Behavior**

Ideally this inline specification should work to control the option. Alternatively, if it can't be supported, it should emit an error to that effect.

For this specific error, I'd probably expect the value specified inline to be additive to the existing values whereas it appears that it (if it worked) it would replace them.

**Investigation**

I stepped through the code a bit and it looks like this is going wrong in `parse_inline_configuration`, probably it should be `Options.apply_changes` that's responsible for this.

The specific issue is that while the `enable_error_codes` attribute (`list[str]`) is updated, `enable_error_codes` (`set[ErrorCode]`, which is what's actually used) isn't updated.

**Comment by PeterJCLaw** [comment]
Continuing my investigation here -- it looks like this is more involved than I thought due to the way that mypy has two places where enabled/disabled error codes are recorded -- on `Options` and again on `Errors`. While `apply_changes` can update the new `Options` instance fairly easily [1], there isn't a great place to update the `Errors` instance that's being used. Moreover it looks like there's only a single `Errors` instance for the whole process, so it's not clear that updating that instance is even the right approach here.

Is it expected that updating the active error codes inline should work?

[1]: a gotcha here is that somehow we need to validate the options and warn the user about incorrect ones, which feels out of scope for the `apply_changes` method -- it feels like that ought to have happened earlier in the processing of the inline options though I don't immediately have a great suggestion of where it should live.

**Comment by PeterJCLaw** [comment]
P.S: I've put an initial change towards this in https://github.com/python/mypy/compare/master...PeterJCLaw:fix-11789-inline-error-code-specification, however that may or not be useful depending on what the right solution is here. It also doesn't address the handling of invalid codes at all.
